### ISSUE TITLE
Validator in the form widget runs twice when submit button pressed when auto validation enabled.

### ISSUE THREAD

**Issue by ishannadeep** [flutter/flutter#80269]
when form widget has autovalidation enabled when pressing submit button cause validator to run twice.It only need to run one time. Below is my submit button.
</details>

```
child: TextButton(
              child: Text("Submit"),
              onPressed: () async {
                await _formKey.currentState?.save();
                await searchContact(_name);
                print("Afte searchContact: $search");
               if (await _formKey.currentState.validate()) {
                  final newContact = Contact(_name, int.parse(_age));
                  if (!search) {
                    addContact(newContact);
                  };
                  clearText();
                }
              }),
```

**Comment by pedromassangocode** [comment]
Hi @ishannadeep 
Can you please provide your `flutter doctor -v`  and a minimal complete reproducible code sample?
Thank you

**Comment by ishannadeep** [comment]
C:\Users\ASUS>flutter doctor -v
[√] Flutter (Channel stable, 2.0.3, on Microsoft Windows [Version 10.0.19042.867], locale en-US)
    • Flutter version 2.0.3 at C:\src\flutter
    • Framework revision 4d7946a68d (4 weeks ago), 2021-03-18 17:24:33 -0700
    • Engine revision 3459eb2436
    • Dart version 2.12.2

[√] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at C:\Users\ASUS\AppData\Local\Android\sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Android Studio (version 4.1.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] Connected device (2 available)
    • Chrome (web) • chrome • web-javascript • Google Chrome 89.0.4389.114
    • Edge (web)   • edge   • web-javascript • Microsoft Edge 89.0.774.75

• No issues found!


` final _formKey = GlobalKey<FormState>();
  final fieldText_name = TextEditingController();
  final fieldText_age = TextEditingController();
  bool search = false;
  String _name;
  String _age;
  //clear the text after submit
  void clearText() {
    fieldText_name.clear();
    fieldText_age.clear();
  }
  //add name and age to database
  addContact(Contact contact) async {
    await Hive.box('contact').add(contact);
  }
 //search the name with database to see if it already has that name
  searchContact(name) async {
    print("searchContact stated :$search");
    var length = Hive.box('contact').length;
    for (int num = 0; num < length; num++) {
      if (name == await Hive.box('contact').getAt(num).name) {
          search = true;
          break;
      }
    }
  }
void dispose(){
    super.dispose();
    fieldText_name.dispose();
    fieldText_age.dispose();
}
  @override
  Widget build(BuildContext context) {
    return Form(
     autovalidateMode: AutovalidateMode.onUserInteraction,
      key: _formKey,
      child: Column(children: <Widget>[
        Row(
          children: <Widget>[
            Expanded(
              child: Wrap(children: [
                TextFormField(
                  controller: fieldText_name,
                  validator: (value1) {
                    RegExp exp1 = RegExp(
                        r"""[0-9\^\`\~\!\@\#\$\%\&\*\(\)\_\-\+\=\{\}\[\]\|\\\:\;\“"’'\<\,\>\.\?\๐\฿\/]""");
                    if (value1 == null || value1.isEmpty) {
                      return 'Empty';
                    } else if (exp1.hasMatch(value1)) {
                      return 'Letters only';
                    } else {
                      print("validation :$search");
                      if (search) {
                        search = false;
                        print("validation inside if :$search");
                        return 'Already has ';
                      }
                    }
                    return null;
                  },
                  decoration: const InputDecoration(
                      labelText: "Name", icon: Icon(Icons.person)),
                  onSaved: (value) => _name = value,
                ),
              ]),
            ),
            Expanded(
              child: Wrap(children: [
                TextFormField(
                  keyboardType: TextInputType.number,
                  controller: fieldText_age,
                  validator: (value) {
                    RegExp exp = RegExp(
                        r"""[a-zA-Z\^\`\~\!\@\#\$\%\&\*\(\)\_\-\+\=\{\}\[\]\|\\\:\;\“"’'\<\,\>\.\?\๐\฿\/]""");
                    RegExp exp2 = RegExp(r'([0-9]\s+[0-9])+');
                    if (value == null || value.isEmpty) {
                      return 'Empty';
                    } else if (exp2.hasMatch(value)) {
                      return "No space ";
                    } else if (exp.hasMatch(value)) {
                      return ' numbers only';
                    } else if (int.parse(value) > 130 ||
                        int.parse(value) == 0) {
                      return 'Not Valid, use (1-130)';
                    } else {
                      return null;
                    }
                  },
                  decoration: InputDecoration(
                      labelText: "Age", icon: Icon(Icons.sd_storage)),
                  onSaved: (value) => _age = value,
                ),
              ]),
            )
          ],
        ),
        Row(
          children: <Widget>[
            Expanded(
              child: TextButton(
                  child: Text("Submit"),
                  onPressed: () async {
                     _formKey.currentState?.save();
                    await searchContact(_name);
                    print("Afte searchContact: $search");
                   if ( _formKey.currentState.validate()) {
                      final newContact = Contact(_name, int.parse(_age));
                      if (!search) {
                        addContact(newContact);
                        clearText();
                        print("setstaet");
                      };
                    }
                  }),
            ),
          ],
        ),
      ]),
    );
  }`

**Comment by pedromassangocode** [comment]
Hi @ishannadeep 
The code you shared seems to be a piece of a project, can you please provide a [minimal complete reproducible code](https://stackoverflow.com/help/minimal-reproducible-example) sample that does not use any third-party plugin/package?
Thank you

**Comment by ishannadeep** [comment]
I've created the minimum reproducible code and found out the problem. I misplaced a variable(search). It was in the validation part. In validation if statement check whether the search variable is true or false. if it is true validation will give the error and reset the variable to default value(False).I had to move it inside the function which check if the input value is already in the database or not.
